#--- ./Makefile~	2005-09-25 22:02:58.000000000 +0300
#+++ ./Makefile	2005-09-25 22:05:35.000000000 +0300
#@@ -240,11 +240,15 @@
# OBJS_MPLAYER += osdep/mplayer-rc.o
# endif
# 
#-$(PRG):	$(MPLAYER_DEP)
#+
#+libmplayer-common.so:
#+	$(CC) $(CFLAGS) -o libmplayer-common.so -shared $(COMMON_LIBS) 
#+
#+$(PRG):	$(MPLAYER_DEP) libmplayer-common.so
#     ifeq ($(TARGET_WIN32),yes)
# 	windres -o osdep/mplayer-rc.o osdep/mplayer.rc
#     endif
#-	$(CC) $(CFLAGS) -o $(PRG) $(OBJS_MPLAYER) libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) $(COMMON_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) $(ARCH_LIB) $(I18NLIBS) $(MATH_LIB)
#+	$(CC) $(CFLAGS) -o $(PRG) -L. -lmplayer-common $(OBJS_MPLAYER) libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) $(ARCH_LIB) $(I18NLIBS) $(MATH_LIB)
# 
# mplayer.exe.spec.c: libmpcodecs/libmpcodecs.a
# 	winebuild -fPIC -o mplayer.exe.spec.c -exe mplayer.exe -mcui \
#@@ -258,8 +262,8 @@
# 	$(CC) $(CFLAGS) -shared -Wl,-Bsymbolic -o mplayer_wine.so mplayer_wine.spec.c $(OBJS_MPLAYER) libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) $(COMMON_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) -lwine $(ARCH_LIB) $(MATH_LIB)
# 
# ifeq ($(MENCODER),yes)
#-$(PRG_MENCODER): $(MENCODER_DEP)
#-	$(CC) $(CFLAGS) -o $(PRG_MENCODER) $(OBJS_MENCODER) libmpcodecs/libmpencoders.a $(ENCORE_LIB) $(COMMON_LIBS) $(EXTRA_LIB) $(MLIB_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(ARCH_LIB) $(I18NLIBS) $(MATH_LIB)
#+$(PRG_MENCODER): $(MENCODER_DEP) libmplayer-common.so
#+	$(CC) $(CFLAGS) -o $(PRG_MENCODER) $(OBJS_MENCODER) -L. -lmplayer-common libmpcodecs/libmpencoders.a $(ENCORE_LIB) $(EXTRA_LIB) $(MLIB_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(ARCH_LIB) $(I18NLIBS) $(MATH_LIB)
# endif
# 
# codecs.conf.h: $(PRG_CFG) etc/codecs.conf
--- ./Makefile~	2005-09-25 22:02:58.000000000 +0300
+++ ./Makefile	2005-09-25 22:06:48.000000000 +0300
@@ -240,11 +240,18 @@
 OBJS_MPLAYER += osdep/mplayer-rc.o
 endif
 
-$(PRG):	$(MPLAYER_DEP)
+
+SHARED_VERSION = 0.0.0
+#	libtool --mode=link $(CC) $(CFLAGS) -o $@ -shared $(COMMON_LIBS) -avoid-version -release $(SHARED_VERSION)
+
+libmplayer.la:
+	libtool --mode=link $(CC) $(CFLAGS) -o $@ -shared $(COMMON_LIBS) -avoid-version -release $(SHARED_VERSION)
+
+$(PRG):	$(MPLAYER_DEP) libmplayer.la
     ifeq ($(TARGET_WIN32),yes)
 	windres -o osdep/mplayer-rc.o osdep/mplayer.rc
     endif
-	$(CC) $(CFLAGS) -o $(PRG) $(OBJS_MPLAYER) libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) $(COMMON_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) $(ARCH_LIB) $(I18NLIBS) $(MATH_LIB)
+	libtool --mode=link $(CC) $(CFLAGS) -o $(PRG) libmplayer.la $(OBJS_MPLAYER) libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) $(ARCH_LIB) $(I18NLIBS) $(MATH_LIB)
 
 mplayer.exe.spec.c: libmpcodecs/libmpcodecs.a
 	winebuild -fPIC -o mplayer.exe.spec.c -exe mplayer.exe -mcui \
@@ -258,8 +265,8 @@
 	$(CC) $(CFLAGS) -shared -Wl,-Bsymbolic -o mplayer_wine.so mplayer_wine.spec.c $(OBJS_MPLAYER) libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) $(COMMON_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) -lwine $(ARCH_LIB) $(MATH_LIB)
 
 ifeq ($(MENCODER),yes)
-$(PRG_MENCODER): $(MENCODER_DEP)
-	$(CC) $(CFLAGS) -o $(PRG_MENCODER) $(OBJS_MENCODER) libmpcodecs/libmpencoders.a $(ENCORE_LIB) $(COMMON_LIBS) $(EXTRA_LIB) $(MLIB_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(ARCH_LIB) $(I18NLIBS) $(MATH_LIB)
+$(PRG_MENCODER): $(MENCODER_DEP) libmplayer.la
+	libtool --mode=link $(CC) $(CFLAGS) -o $(PRG_MENCODER) $(OBJS_MENCODER) libmplayer.la libmpcodecs/libmpencoders.a $(ENCORE_LIB) $(EXTRA_LIB) $(MLIB_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(ARCH_LIB) $(I18NLIBS) $(MATH_LIB)
 endif
 
 codecs.conf.h: $(PRG_CFG) etc/codecs.conf
